File: anidesk_py/src/anidesk/services/cloud.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Protocol


@dataclass(slots=True)
class CloudVersion:
    id: str
    name: str
    updated_at: str
    size: int


class CloudSyncProvider(Protocol):
    def connect(self) -> None: ...
    def disconnect(self) -> None: ...
    def upload(self, name: str, data: bytes) -> CloudVersion: ...
    def download(self, version_id: str) -> bytes: ...
    def list_versions(self) -> list[CloudVersion]: ...


class CloudSyncNotConfigured:
    @staticmethod
    def _unavailable() -> None:
        raise RuntimeError("云同步将在第二阶段接入，目前请使用本地备份。")

    def connect(self) -> None: self._unavailable()
    def disconnect(self) -> None: return None
    def upload(self, name: str, data: bytes) -> CloudVersion: self._unavailable(); raise AssertionError
    def download(self, version_id: str) -> bytes: self._unavailable(); raise AssertionError
    def list_versions(self) -> list[CloudVersion]: self._unavailable(); raise AssertionError

File: src/services/cloud.ts
export interface CloudVersion {
  id: string;
  name: string;
  updatedAt: string;
  size: number;
}

export interface CloudSyncProvider {
  connect(): Promise<void>;
  disconnect(): Promise<void>;
  upload(name: string, data: Uint8Array): Promise<CloudVersion>;
  download(versionId: string): Promise<Uint8Array>;
  listVersions(): Promise<CloudVersion[]>;
}

export class CloudSyncNotConfigured implements CloudSyncProvider {
  private unavailable(): never {
    throw new Error("云同步将在第二阶段接入，目前请使用本地备份。");
  }
  async connect(): Promise<void> { this.unavailable(); }
  async disconnect(): Promise<void> { return; }
  async upload(_name: string, _data: Uint8Array): Promise<CloudVersion> { return this.unavailable(); }
  async download(_versionId: string): Promise<Uint8Array> { return this.unavailable(); }
  async listVersions(): Promise<CloudVersion[]> { return this.unavailable(); }
}
